src/components/cards/cards.jsx

```diff
@@ -0,0 +1,284 @@
+                video: PropTypes.string,
+                deckIds: PropTypes.arrayOf(PropTypes.string)
+            }))
+        })
+    }).isRequired,
+    deckIds: PropTypes.arrayOf(PropTypes.string).isRequired,
+    onActivateDeckFactory: PropTypes.func.isRequired,
+    onShowAll: PropTypes.func.isRequired
+};
+
+const Cards = props => {
+    if (props.activeDeckId === null) return;
+
+    const steps = props.content[props.activeDeckId].steps;
+
+    return (
+        <Draggable
+            bounds="parent"
+            position={{x: props.x, y: props.y}}
+            onDrag={props.onDrag}
+            onStart={props.onStartDrag}
+            onStop={props.onEndDrag}
+        >
+            <div className={styles.cardContainer}>
+                <div className={styles.card}>
+                    <CardHeader
+                        step={props.step}
+                        totalSteps={steps.length}
+                        onCloseCards={props.onCloseCards}
+                        onShowAll={props.onShowAll}
+                    />
+                    <div className={styles.stepBody}>
+                        {steps[props.step].deckIds ? (
+                            <PreviewsStep
+                                content={props.content}
+                                deckIds={steps[props.step].deckIds}
+                                onActivateDeckFactory={props.onActivateDeckFactory}
+                                onShowAll={props.onShowAll}
+                            />
+                        ) : (
+                            steps[props.step].video ? (
+                                <VideoStep
+                                    dragging={props.dragging}
+                                    video={steps[props.step].video}
+                                />
+                            ) : (
+                                <ImageStep
+                                    image={steps[props.step].image}
+                                    title={steps[props.step].title}
+                                />
+                            )
+                        )}
+                    </div>
+                    <NextPrevButtons
+                        onNextStep={props.step < steps.length - 1 ? props.onNextStep : null}
+                        onPrevStep={props.step > 0 ? props.onPrevStep : null}
+                    />
+                </div>
+            </div>
+        </Draggable>
+    );
+};
+
+Cards.propTypes = {
+    activeDeckId: PropTypes.string.isRequired,
+    content: PropTypes.shape({
+        id: PropTypes.shape({
+            name: PropTypes.node.isRequired,
+            img: PropTypes.string.isRequired,
+            steps: PropTypes.arrayOf(PropTypes.shape({
+                title: PropTypes.node,
+                image: PropTypes.string,
+                video: PropTypes.string,
+                deckIds: PropTypes.arrayOf(PropTypes.string)
+            }))
+        })
```

Comment:
It would be easy to make a mistake and modify content without React noticing, so since only `content[activeDeckId]` is used, it would be more error-proof to pass in `name`, `img` and `steps` directly (easy to do with a spread like `{...content[activeDeckId}`).